From dcfdaad239372516a4f4826349afe4e613aaedec Mon Sep 17 00:00:00 2001
From: David Yu-Chia Chang <dchang@novell.com>
Date: Thu, 19 Aug 2010 10:38:32 +0200
Subject: Bug 594088 - Palimpsest GUI impossible to use on small screens
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Add scrollbars to the right pane if the window is too small to fit the entire
content.

Signed-off-by: Martin Pitt <martin.pitt@ubuntu.com>
---
 src/palimpsest/gdu-shell.c |    7 ++++++-
 1 files changed, 6 insertions(+), 1 deletions(-)

Index: gnome-disk-utility/src/palimpsest/gdu-shell.c
===================================================================
--- gnome-disk-utility.orig/src/palimpsest/gdu-shell.c	2011-05-19 23:30:02.256420106 +0200
+++ gnome-disk-utility/src/palimpsest/gdu-shell.c	2011-05-19 23:30:12.636289615 +0200
@@ -1097,6 +1097,7 @@
         GtkAccelGroup *accel_group;
         GtkWidget *hpane;
         GtkWidget *tree_view_scrolled_window;
+        GtkWidget *hpane_right_scrolled_window;
         GtkTreeSelection *select;
         GtkWidget *label;
         GtkTreeViewColumn *column;
@@ -1154,7 +1155,11 @@
 
         shell->priv->sections_vbox = gtk_vbox_new (FALSE, 12);
         gtk_container_set_border_width (GTK_CONTAINER (shell->priv->sections_vbox), 6);
-        gtk_box_pack_start (GTK_BOX (vbox2), shell->priv->sections_vbox, TRUE, TRUE, 0);
+        hpane_right_scrolled_window = gtk_scrolled_window_new (NULL, NULL);
+        gtk_scrolled_window_set_policy (GTK_SCROLLED_WINDOW (hpane_right_scrolled_window), 
+                                        GTK_POLICY_AUTOMATIC, GTK_POLICY_AUTOMATIC);
+        gtk_scrolled_window_add_with_viewport(hpane_right_scrolled_window, shell->priv->sections_vbox);
+        gtk_box_pack_start (GTK_BOX (vbox2), hpane_right_scrolled_window, TRUE, TRUE, 0);
 
         /* setup and add horizontal pane */
         hpane = gtk_hpaned_new ();
